Sources/Datadog/RUM/RUMVitals/VitalRefreshRateReader.swift

```diff
@@ -53,25 +54,47 @@ internal class VitalRefreshRateReader: ContinuousVitalReader {
         }
     }
 
-    // MARK: - Private
+    // MARK: - Internal
 
     @objc
-    private func displayTick(link: CADisplayLink) {
-        if let lastTimestamp = self.lastFrameTimestamp {
-            let frameDuration = link.timestamp - lastTimestamp
-            let currentFPS = 1.0 / frameDuration
-
-            for publisher in valuePublishers {
-                publisher.mutateAsync { currentInfo in
-                    currentInfo.addSample(currentFPS)
-                }
+    internal func displayTick(link: CADisplayLink) {
+        guard let fps = framesPerSecond(provider: link) else {
+            return
+        }
+
+        for publisher in valuePublishers {
+            publisher.mutateAsync { currentInfo in
+                currentInfo.addSample(fps)
             }
         }
-        lastFrameTimestamp = link.timestamp
     }
 
+    internal func framesPerSecond(provider: FrameInfoProvider) -> Double? {
+        var fps: Double? = nil
+
+        if let lastFrameTimestamp = self.lastFrameTimestamp {
+            let currentFrameDuration = provider.currentFrameTimestamp - lastFrameTimestamp
+            let currentFPS = 1.0 / currentFrameDuration
+
+            // ProMotion displays (e.g. iPad Pro and newer iPhone Pro) can have refresh rate higher than 60 FPS.
+            if let expectedCurrentFrameDuration = self.nextFrameDuration, provider.maximumDeviceFramesPerSecond > 60 {
```

Comment:
We could export `60` to some property called i.e `maximumBackendSupportedFrameRate` just to remember the reason in the code